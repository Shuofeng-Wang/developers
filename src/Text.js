import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { space, typography, textStyle, color, layout } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";

// const TextComponent= styled.text`
//     text-transform: ${props => props.textTransform || "none"};
//     text-decoration-line: ${props => props.textDecorationLine || "none"};
//     ${space};
//     ${typography};
//     ${textStyle};
//     ${color};
//     ${layout};
// `;

const TextComponent = styled("div").withConfig({
    shouldForwardProp: (prop, defaultValidatorFn) =>
        !["color", "fontSize", "fontStyle", "textAlign", "textTransform", "textDecorationLine", "m", "fontWeight"].includes(prop),
})`
  text-transform: ${props => props.textTransform || "none"};
  text-decoration-line: ${props => props.textDecorationLine || "none"};
  ${space};
  ${textStyle};
  ${typography};
  ${layout};
  ${color};
`;

const Text = ({ textTransform, textDecorationLine, ...props }) => {
    return <TextComponent textTransform={textTransform} textDecorationLine={textDecorationLine} {...props} />;
};

// const Text = props => {
//   console.log("PROPS: ", props);
//   const { textTransform, textDecorationLine, ...other } = props;
//
//   console.log("MyTransform: ", textTransform);
//   console.log("MyDecorationLine: ", textDecorationLine);
//
//   return <TextComponent textTransform={textTransform}  textDecorationLine={textDecorationLine} {...other} />;
// }

Text.displayName = "Text";

Text.defaultProps = {
  as: "div",
  color: "currentColor",
  fontSize: "30px",
  fontStyle: "normal",
  textAlign: "left",
  textTransform: "none",
  textDecorationLine: "none",
};

Text.propTypes = {
  ...styledProps.space,
  ...styledProps.typography,
  ...styledProps.textStyle,
  ...styledProps.color,
  ...styledProps.layout,
  color: PropTypes.string,
  textTransform: PropTypes.oneOf(["capitalize", "uppercase", "lowercase", "none"]),
  textDecorationLine: PropTypes.oneOf(["overline", "line-through", "underline", "none"]),
};

export default Text;
