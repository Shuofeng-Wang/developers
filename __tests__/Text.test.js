import React from "react";
import renderer from "react-test-renderer";
import Text from "../src/Text";

test("Text renders correctly", () => {
    const tree = renderer.create( <Text> Something{" "} </Text> ).toJSON();
    expect(tree).toMatchSnapshot();
});

test("Text renders fontStyle correctly", () => {
    const tree2 = renderer.create( <Text italic> Something </Text> ).toJSON();
    expect(tree2).toMatchSnapshot();
});
