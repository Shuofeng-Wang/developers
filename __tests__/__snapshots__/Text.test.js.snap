// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<div
  className="Text__TextElement-sc-102lho-0 glCa-di"
>
   Something
   
   
</div>
`;

exports[`Text renders fontStyle correctly 1`] = `
<div
  className="Text__TextElement-sc-102lho-0 jBaQIo"
>
   Something 
</div>
`;
